src/main/java/de/tum/in/www1/artemis/web/rest/metis/PostResource.java

```diff
@@ -63,6 +63,38 @@ public PostResource(PostService postService) {
         return new ResponseEntity<>(updatedPost, null, HttpStatus.OK);
     }
 
+    /**
+     * PUT /courses/{courseId}/posts/{postId}/pin : Pin an existing post by setting the pinned flag to true (is used as criterion for sorting the posts)
+     *
+     * @param courseId  id of the course the post belongs to
+     * @param postId    id of the post to pin
+     * @param pinState  updated boolean value of the pinned flag for the given post
+     * @return ResponseEntity with status 200 (OK) containing the updated post in the response body,
+     * or with status 400 (Bad Request) if the checks on user, course or post validity fail
+     */
+    @PutMapping("courses/{courseId}/posts/{postId}/pin")
+    @PreAuthorize("hasRole('TA')")
+    public ResponseEntity<Post> updatePinState(@PathVariable Long courseId, @PathVariable Long postId, @RequestBody Boolean pinState) {
+        Post postWithUpdatedPinState = postService.updatePinState(courseId, postId, pinState);
```

Comment:
Should this be wrapped in something like
```
if (authorizationCheckService.isAtLeastTeachingAssistantInCourse(course, user)) {
```
? I think right now every user with `TA` permissions in any course can call this.

Reply:
Or directly `checkHasAtLeastRoleInCourseElseThrow(Role.TEACHING_ASSISTANT, course, user);` which will throw an error with a good message.

Reply:
The check is done in the `PostService` ( I know that this is different in other resources, but for now, I would like to keep it as it is. If at all, I want to refactor the whole `PostResource` in order to move back the checks (authorization etc.) back to the resource.
[Here are the lines](https://github.com/ls1intum/Artemis/pull/3907/files#diff-f29513fa2181af304bfb19fc08e3533d729b117825f94f21131ca0f204b9b610R101-R104) you were looking for.

Reply:
You are right. in the `updatePost` method.